File: Códigos_Celis/Rust/src/busqueda_binaria.rs
// Resolver f(x) = x^3 - x - 2 usando bisección

fn f(x: f64) -> f64 {
    x.powi(3) - x - 2.0
}

fn main() {
    let mut a = 1.0_f64;
    let mut b = 2.0_f64;
    let tolerancia = 1e-6_f64;
    let mut error = 1.0_f64;
    let mut iteracion = 0u32;
    let mut c = 0.0_f64;

    while error > tolerancia {
        c = (a + b) / 2.0;

        let fa = f(a);
        let fc = f(c);

        println!("Iteración: {}", iteracion);
        println!("a = {} b = {} c = {}", a, b, c);
        println!("f(c) = {}", fc);
        println!("-----------------------");

        if fa * fc < 0.0 {
            b = c;
        } else {
            a = c;
        }

        error = (b - a).abs();
        iteracion += 1;
    }

    println!("Raíz aproximada: {}", c);
}

File: Codigos_Celis/Python/BusquedaBinaria.py
# Resolver f(x) = x^3 - x - 2 usando bisección

a = 1
b = 2
tolerancia = 1e-6
error = 1
iteracion = 0

while error > tolerancia:

    c = (a + b) / 2

    fa = a ** 3 - a - 2
    fc = c ** 3 - c - 2

    print("Iteración:", iteracion)
    print("a =", a, "b =", b, "c =", c)
    print("f(c) =", fc)
    print("-----------------------")

    if fa * fc < 0:
        b = c
    else:
        a = c

    error = abs(b - a)
    iteracion += 1

print("Raíz aproximada:", c)